cloud/scope/powervs_cluster.go

```diff
@@ -840,7 +840,7 @@ func (s *PowerVSClusterScope) ReconcileNetwork() (bool, error) {
 	if networkID != nil {
 		s.V(3).Info("Found PowerVS network in IBM Cloud", "id", networkID)
 		s.SetStatus(infrav1beta2.ResourceTypeNetwork, infrav1beta2.ResourceReference{ID: networkID, ControllerCreated: ptr.To(false)})
-		return false, nil
+		return true, nil
```

Comment:
what will happen when network is there not not in ready state?

Reply:
Are you asking for not ready state for DHCP server? If so we never check for ready state of DHCP server when ever user provides an existing DHCP server name, We try to formulate the possible network name from DHCP server name : https://github.com/Karthik-K-N/cluster-api-provider-ibmcloud/blob/3eb2b524c65167565c595d01e475497b066ef22a/cloud/scope/powervs_cluster.go#L884-L891

I think we should check for DHCP server state as well but not sure there will be condition where DHCP server is not ready but still has network.

Reply:
Also @Amulyam24 this ReconcileNetwork is not following the format of other reconcile function, In every other we expect requeue and error but here we are looking for DHCPactive or err, Thats the main reason for this uncaught error.

Reply:
> Also @Amulyam24 this ReconcileNetwork is not following the format of other reconcile function, In every other we expect requeue and error but here we are looking for DHCPactive or err, Thats the main reason for this uncaught error.

@Karthik-K-N, this was changed recently via https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1869
cc @dharaneeshvrd

Reply:
Since we are not requeuing based on ReconcileNetwork return values, made the changes to set NetworkReady condition based on DHCP server active state returned from ReconcileNetwork. 
Missed to handle the existing DHCP server logic, with this change it would return true and NetworkReady can be set to true. 
Agree and we should check the state of DHCP server and decide the DHCP server is ok or not based on that.
Because underneath its just a VM, there is a possibility for network to be created and VM can go into bad state.

Reply:
ok then shall I create a issue to track this?

Reply:
Ref. https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/1933